### PR TITLE
feat: create a reusable drawer for delete confirmation

### LINKED ISSUE #119: Create a reusable drawer for delete confirmation
## Description
The delete confirmation drawer is used in three ways. One is to delete a landlord's account, the second is to delete a sharehouse, and the other is to delete a category. The structure is almost the same, so we'll implement a reusable drawer for such use cases. You don't need to implement the actual business logic to delete account/sharehouse/category yet.

image

## Criteria

- [ ] Utilize pre-created common components like `Drawer`, `Checkbox`, and `Button`
- [ ] Display a relevant warning message to be sure
- [ ] Add a checkbox to confirm with the user if they really want to perform that action
- [ ] Add a `Delete` button to execute the deletion
- [ ] Disable the `Delete` button when the checkbox is not checked 
- [ ] Add a `Cancel` button to close the drawer (cancel the action) 
- [ ] Make sure the component can perform dynamically with props
- [ ] Make sure it can accept a function that executes when the `Delete` button is clicked

### PR DESCRIPTION
**Pull request by yuki-92c**
## Overview

- I've implemented `delete confirmation` drawer components

## Changes

- created a file `src/components/ui/drawers/DeleteConfirmation.tsx`

## Review points

- [ ] Disable the Delete button when the checkbox is not checked
- [ ] Close the drawer by `Cancel` button 
- [ ] the component can perform dynamically with props

## Screenshots or videos

https://github.com/Tascurator/tascurator-frontend/assets/58718006/a350963a-0c61-436d-bea1-895bf22c213a

## Notes

- sample code

```ts

import {DeleteConfirmation } from '@/components/ui/drawers/DeleteConfirmation';
import { useState } from 'react';
import { Button } from '@/components/ui/button';

  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
  const deleteItem = 'MY DELETE ITEM';

  return (
        <Button onClick={() => setIsDrawerOpen(true)}>Delete</Button>
        <DeleteConfirmation 
        deleteItem={deleteItem}
        open={isDrawerOpen} setOpen={setIsDrawerOpen}/>

```

## Assignee Checklist:

- [ ] The base branch is correct (no accidental merges)
- [ ] The branch name follows our branch naming rules
- [ ] The PR title follows our PR title rules
- [ ] My code follows our coding style
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [ ] Code readability and simplicity
- [ ] Follows best practices and coding standards
- [ ] Understandable and maintainable code

### CODE REVIEW

File: src/components/ui/drawers/DeleteConfirmation.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            />
+            <p>{`Yes, I want to delete ${deleteItem}.`}</p>
+          </div>
+        </DrawerDescription>
+        <DrawerFooter>
+          <div className="flex-1">
+            <DrawerClose asChild>
+              <Button
+                type={'button'}
+                variant={'secondary'}
```
Comment (HrRn) [inline src/components/ui/drawers/DeleteConfirmation.tsx]: Could you fix the button variant from secondary to outline?

File: src/components/ui/drawers/DeleteConfirmation.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  return (
+    <Drawer open={open} onOpenChange={setOpen}>
+      <DrawerTrigger />
+      <DrawerContent>
+        <DrawerClose />
+        <DrawerTitle>Confirm delete</DrawerTitle>
+        <DrawerDescription>
+          {`Are you sure you want to delete ${deleteItem}?`}
+          <div className="flex items-center content-start pt-4">
+            <Checkbox
```
Comment (HrRn) [inline src/components/ui/drawers/DeleteConfirmation.tsx]: IMO, it would be better if the <p> tag could also be clicked using the <label> tag.

**Review by HrRn — CHANGES_REQUESTED** [review]

**Comment by Akikaze1119** [comment]
Could you please add 'Drawer' to the end of the file and component names?
DeleteConfirmation.tsx
　↓
DeleteConfirmationDrawer.tsx

**Review by Akikaze1119 — APPROVED** [review]
LGTM:)
Thank you!!

**Review by HrRn — APPROVED** [review]
LGTM :)

**Pull request merged**